packages/material-ui/src/IconButton/IconButton.js

```diff
@@ -27,76 +66,85 @@ export const styles = (theme) => ({
         backgroundColor: 'transparent',
       },
     },
-    '&$disabled': {
-      backgroundColor: 'transparent',
-      color: theme.palette.action.disabled,
-    },
-  },
-  /* Styles applied to the root element if `edge="start"`. */
-  edgeStart: {
-    marginLeft: -12,
-    '$sizeSmall&': {
-      marginLeft: -3,
-    },
-  },
-  /* Styles applied to the root element if `edge="end"`. */
-  edgeEnd: {
-    marginRight: -12,
-    '$sizeSmall&': {
-      marginRight: -3,
-    },
-  },
-  /* Styles applied to the root element if `color="inherit"`. */
-  colorInherit: {
-    color: 'inherit',
-  },
-  /* Styles applied to the root element if `color="primary"`. */
-  colorPrimary: {
-    color: theme.palette.primary.main,
-    '&:hover': {
-      backgroundColor: alpha(theme.palette.primary.main, theme.palette.action.hoverOpacity),
-      // Reset on touch devices, it doesn't add specificity
-      '@media (hover: none)': {
-        backgroundColor: 'transparent',
+  }),
+  ({ styleProps }) => ({
```

Comment:
Do we need a different block for it?

```suggestion
```

Reply:
That was what @mnajdova had [requested](https://github.com/mui-org/material-ui/pull/24542#discussion_r562444841), yeah.

Reply:
It was for this line https://github.com/mui-org/material-ui/pull/24542/files#r562444841 as we have `"&:hover"` styles there, it is not necessary everywhere. Agree, we can simplify